### PR TITLE
[brpc] Updata to 1.1.0

### LINKED ISSUE #24974: [brpc] update to 1.1.0
Library name:

brpc

New version number:

1.1.0

Other information that may be useful (release notes, etc...)

https://github.com/apache/incubator-brpc/releases/tag/1.1.0

### PR DESCRIPTION
**Pull request by day253**
Fixes #24974

Update brpc to the latest version 1.1.0

Tested successfully in the following triplet:
 - x64-linux

### CODE REVIEW

**Comment by day253** [comment]
@JackBoosY 

I can build this port locally. But the azure ci shows below.

```bash
CMake Error at ports/brpc/portfile.cmake:24 (vcpkg_cmake_config_fixup):
  Unknown CMake command "vcpkg_cmake_config_fixup".
Call Stack (most recent call first):
  scripts/ports.cmake:146 (include)
```

**Comment by day253** [comment]
@JackBoosY this pr is ready.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/brpc/fix-build.patch [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+@@ -197,9 +202,10 @@ set(DYNAMIC_LIB
+     ${LEVELDB_LIB}
+     ${PROTOC_LIB}
+     ${CMAKE_THREAD_LIBS_INIT}
+-    ${THRIFT_LIB}
+-    ${THRIFTNB_LIB}
+     ${OPENSSL_CRYPTO_LIBRARY}
++    ${OPENSSL_SSL_LIBRARY}
++    ${THRIFT_LIB}
++    # ${THRIFTNB_LIB}
```
Comment (ras0219-msft) [inline ports/brpc/fix-build.patch:80]:
```suggestion
```

File: ports/brpc/fix-build.patch [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
++foreach (TARGET_FILE ${TARGET_FILES})
++    include("${TARGET_FILE}")
++endforeach()
++]])
++install(FILES "${CMAKE_CURRENT_BINARY_DIR}/brpc-config.cmake" DESTINATION share/brpc)
++
++install(EXPORT brpcTargets
++    NAMESPACE brpc::
++    DESTINATION share/brpc
++)
```
Comment (ras0219-msft) [inline ports/brpc/fix-build.patch:222]: These need to be under the `unofficial::brpc::` namespace and in an `unofficial-brpc` package:

https://vcpkg.io/en/docs/maintainers/maintainer-guide.html#add-cmake-exports-in-an-unofficial--namespace

File: ports/brpc/fix-build.patch [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
++endif()
++
++# # for protoc-gen-mcpack
++# set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/output/bin)
++
++# set(protoc_gen_mcpack_SOURCES
++#     ${PROJECT_SOURCE_DIR}/src/mcpack2pb/generator.cpp
++#  )
++# add_executable(protoc-gen-mcpack ${protoc_gen_mcpack_SOURCES})
++# target_link_libraries(protoc-gen-mcpack brpc-shared)
```
Comment (ras0219-msft) [inline ports/brpc/fix-build.patch:174]:
```suggestion
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Review by ras0219-msft — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**